### ISSUE TITLE
Cargo install's new updating feature doesn't pull from git repo.

### ISSUE THREAD

**Issue by Amanita-muscaria** [rust-lang/cargo#7852]
**Problem**
Cargo's new updating feature doesn't pull from git repo.

**Steps**

1. `cargo install --git https://github.com/nushell/nushell.git nu --features stable` make it point at something before HEAD.
2. `cargo install nu` 
3. cargo will pull from crates.io instead of the git repo.

**Possible Solution(s)**
I know that nu is on crates.io but as I'm actively working on it (and also I love it) I wanted to pull from the repo so as to get all the bugs as I'm using it. It could be that I'm just using this feature incorrectly.

**Notes**

Output of `cargo version`:
cargo 1.41.0 (626f0f40e 2019-12-03)

**Comment by ehuss** [comment]
This is intentional.  You need to specify the source each time you run `cargo install`, otherwise it wouldn't know whether or not you wanted to switch sources.  That is, you need to specify `--git` each time you want to install from the git repo.

**Comment by Amanita-muscaria** [comment]
Do I have to include the url or just --git?

**Comment by ehuss** [comment]
The command more or less needs to be the same (`--git` with url, the same features, etc.).

**Comment by Amanita-muscaria** [comment]
Oh, that's not terrible ergonomic. If someone could point me in the right direction, I could probably implement a flag or something to indicate where to update from.

**Comment by ehuss** [comment]
It sounds like you're looking for a new command for managing installed packages.  I think something like that would require an RFC process to work on the design.  There are also third-party plugins like [cargo-update](https://crates.io/crates/cargo-update), which I think might do something similar.

**Comment by Amanita-muscaria** [comment]
No, not necessarily a new command, but a flag for the existing install command makes sense to me. If that's out of scope or something, it's no big deal, I can just use the cargo-update plugin like you suggested.
